test/e2e/kube-ovn/underlay/underlay.go

```diff
@@ -1251,6 +1251,81 @@ var _ = framework.SerialDescribe("[group:underlay]", func() {
 			framework.ExpectEmpty(conflictVlanSubnet2.Status.V6AvailableIPRange)
 		}
 	})
+
+	framework.ConformanceIt("should support nodeSelector to include only specific nodes", func() {
+		f.SkipVersionPriorTo(1, 15, "This feature was introduced in v1.15")
+
+		ginkgo.By("Getting k8s nodes")
+		k8sNodes, err := e2enode.GetReadySchedulableNodes(context.Background(), cs)
+		framework.ExpectNoError(err)
+		framework.ExpectNotEmpty(k8sNodes.Items)
+
+		// Select the first node for inclusion
+		selectedNodeName := k8sNodes.Items[0].Name
+		testLabelKey := "provider-network-test"
+		testLabelValue := "selected"
+
+		ginkgo.By("Adding test label to selected node " + selectedNodeName)
+		selectedNode := &k8sNodes.Items[0]
+		if selectedNode.Labels == nil {
+			selectedNode.Labels = make(map[string]string)
+		}
+		selectedNode.Labels[testLabelKey] = testLabelValue
+		_, err = cs.CoreV1().Nodes().Update(context.Background(), selectedNode, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Creating provider network with nodeSelector " + providerNetworkName)
+		pn := makeProviderNetwork(providerNetworkName, false, linkMap)
+		pn.Spec.NodeSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				testLabelKey: testLabelValue,
```

Comment:
```suggestion
				corev1.LabelHostname: selectedNode.Annotations[corev1.LabelHostname],
```

We can use existing label `kubernetes.io/hostname`.

Reply:
It's more flexible if we need to test a group of nodes.